File: Core/Entities/Post.cs
namespace Core.Entities
{
    /* 
    The Post entity, representing the core of the application, the post. Includes the category, steps, user and likes that are attributed to the post
    */
    public class Post : BaseEntity
    {
        public string Title { get; set; }
        public string Description { get; set; }
        public string Picture { get; set; }
        public string CategoryId { get; set; }
        public Category Category { get; set; }
        public ICollection<Step> Steps { get; set; }
        public string UserId { get; set; }
        public AppUser User { get; set; }
        public ICollection<Like> Likes { get; set; }
        public bool IsLiked { get; set; } = false;
    }
}

File: client/src/app/shared/models/post.ts
import { Step } from './step';

// Interface for the posts retrieved from the backend
export interface Post {
  id: string;
  title: string;
  description: string;
  picture: string;
  category: string;
  user: string;
  likes: number;
  isLiked: boolean;
  steps: Step[];
}

export interface PostFormData {
  title?: string | null;
  description?: string | null;
  picture?: string | null;
  categoryId?: string | null;
  steps?: Step[];
}
